tests/src/test/spark330/scala/org/apache/spark/sql/rapids/utils/RapidsSQLTestsBaseTrait.scala

```diff
@@ -107,7 +116,36 @@ trait RapidsSQLTestsBaseTrait extends SharedSparkSession with RapidsTestsBaseTra
   }
 }
 
-object RapidsSQLTestsBaseTrait {
+object RapidsSQLTestsBaseTrait extends Logging {
+  private val resourceMap = scala.collection.mutable.Map.empty[String, java.nio.file.Path]
+  private val jarUrlRegex = raw"jar:file:(/.*)!.*".r
+  TrampolineUtil.addShutdownHook(10000, () => {
+    resourceMap.valuesIterator.foreach { dirPath =>
+      logWarning(s"Deleting expanded test jar dir $dirPath")
+      FileUtil.fullyDelete(dirPath.toFile)
+    }
+  })
+
+  private def expandJar(jarPath: String): java.nio.file.Path = {
+    val jarFile = new java.io.File(jarPath)
+    val destDir = java.nio.file.Files.createTempDirectory(jarFile.getName + ".expanded")
+    logWarning(s"Registering $destDir for deletion on exit")
+    FileUtil.unZip(jarFile, destDir.toFile)
+    destDir
+  }
+
+  def sparkTestResourcesDir(testClass: Class[_]): java.nio.file.Path = {
+    val sparkTestClassResource = "/" + testClass.getSuperclass.getName.replace(".", "/") + ".class"
```

Comment:
Is just getting the superclass a safe assumption? We have some RAPIDS suite classes that derive from other RAPIDS classes before they derive from a Spark suite class.   Should this be looking up the class hierarchy for a  class without ".rapids" in the fully qualified classname?

Reply:
Good point, this works only for classes immediately deriving from a Suite in Spark repo